File: gwt/src/core/util/InternalResource.java
/**
 * Author: Timothy Prepscius
 * License: GPLv3 Affero + keep my name in the code!
 */
package core.util;

import java.io.ByteArrayInputStream;
import java.io.InputStream;

public class InternalResource
{
	public static native String getResource(String r) /*-{
		return $wnd.resource_acquire(r);
	}-*/;
	
	public static InputStream getResourceAsStream (Class<?> c, String p)
	{
		return new ByteArrayInputStream(Base64.decode(getResource(c.getName() + "." + p)));
	}

}

File: cpp/core/mailiverse/core/util/InternalResource.cpp
/**
 * Author: Timothy Prepscius
 * License: BSD + keep my name in the code!
 */

#include "InternalResource.h"
#include "../crypt/Base64.h"

using namespace mailiverse::core::util;
using namespace mailiverse::core::crypt;
using namespace mailiverse::core;
using namespace mailiverse;

Block InternalResource::get(const String &package, const String &subKey)
{
	String key = package + "." + subKey;
	
	if (package=="mailiverse::mail::manager::Mailer" && subKey == "truststore.jks")
	{
		return toBlock(
			"YOUR TRUST STORE"
		);
	}
	if (package=="mailiverse::mail::manager::Notifications" && subKey == "truststore.jks")
	{
		return toBlock (
			"YOUR TRUST STORE"
		);
	}
	
	return Block();
}